### PR TITLE
Add changelog target to make

### PR DESCRIPTION
**Pull request by pavolloffay**
I have written https://github.com/pavolloffay/github-changelog, I could not find anything good and customizable. Now it generates only non categorized changes. I will extend it to create categories based on labels on PRs.

```
# Changelog
 
* Use latest all-in-one image in hotrod ([#1555](https://github.com/jaegertracing/jaeger/pull/1555), [@everett980](https://github.com/everett980))
* Add CA certs to all-in-one image ([#1554](https://github.com/jaegertracing/jaeger/pull/1554), [@chandresh-pancholi](https://github.com/chandresh-pancholi))
* temporarily expose grpc conn ([#1547](https://github.com/jaegertracing/jaeger/pull/1547), [@guanw](https://github.com/guanw))
 
## v1.12.0 (2019-05-16)
* Add all storages to env command ([#1541](https://github.com/jaegertracing/jaeger/pull/1541), [@pavolloffay](https://github.com/pavolloffay))
* Prepare release 1.12.0 ([#1538](https://github.com/jaegertracing/jaeger/pull/1538), [@pavolloffay](https://github.com/pavolloffay))
* Bump UI to 1.2.0 ([#1536](https://github.com/jaegertracing/jaeger/pull/1536), [@pavolloffay](https://github.com/pavolloffay))
* Update python script shebang to explicitly reference used python vers… ([#1534](https://github.com/jaegertracing/jaeger/pull/1534), [@objectiser](https://github.com/objectiser))
* Memory store limit queries return most recent traces ([#1394](https://github.com/jaegertracing/jaeger/pull/1394), [@jacobmarble](https://github.com/jacobmarble))
* Add grpc_resolver using external discovery service ([#1498](https://github.com/jaegertracing/jaeger/pull/1498), [@guanw]
```
Full changelog https://pastebin.com/b8ezHg4s

### CODE REVIEW

**Comment by objectiser** [comment]
Good start - I think the first labels that will be required should indicate internal project PRs and doc PRs that we wouldn't necessarily want included in the changelog. Then the next obvious would be breaking changes. 

Is your intention to make the tool configurable so that a list of included and excluded labels could be provided? and then the PRs would be subdivided into those included labels?

Possibly PRs that don't have one of the included/exluded labels should be flagged as ambiguous? to prevent some falling through the cracks.

**Comment by pavolloffay** [comment]
What labels will be rendered also depends on the template. You could write a template which does not render  `bug`, `trivial` labels. We could also have a flag to exclude some labels from the commit list completely. 

If you have more suggestion please open issue here https://github.com/pavolloffay/github-changelog/issues.

**Comment by pavolloffay** [comment]
The currently generated changelog just simplifies the work a person doing the release has to do. The commits still have to be manually categorized.

File: RELEASE.md [hunk 1]
```diff
@@ -2,8 +2,9 @@
 
 1. Create a PR "Preparing release X.Y.Z" against master or maintenance branch ([example](https://github.com/jaegertracing/jaeger/pull/543/files)) by updating CHANGELOG.md to include:
     * A new section with the header `<X.Y.Z> (YYYY-MM-DD)`
-    * A curated list of notable changes and links to PRs. Do not simply dump git log, select the changes that affect the users.
     * The section can be split into sub-section if necessary, e.g. UI Changes, Backend Changes, Bug Fixes, etc.
+    * Run `make changelog` and manually select notable changes that affect users. The incorrect pull request titles can be changed on Github and changelog regenerated.
+    * A curated list of notable changes and links to PRs. Do not simply dump git log, select the changes that affect the users. To obtain the list of all changes run `make changelog`
```
Comment (objectiser) [inline RELEASE.md]: I think this bullet point could be moved back to its original position with the new part at the end. The previous new bullet point I think is redundant.

**Comment by yurishkuro** [comment]
Cool!

* how does the tool know which changes to pick from git log? When I ran it I got a LOT of output. I've seen other tools take a reference/tag like `-since v1.12.0`
* when I run make target I am getting a warning `make: Circular changelog <- changelog dependency dropped.`
* the tool is adding credits using only github handle `[@pavolloffay](https://github.com/pavolloffay)` - it would be nice to include the full name, which is already recorded in git log

**Comment by pavolloffay** [comment]
Now it parses full github history, since/until will be supported later https://github.com/pavolloffay/github-changelog/issues/6

Credits - we could change to full name, but so far I have seen using only handles as those do not change.


I have also created a couple of labels with `changelog` prefix to match our current categories. We could start using them on PRs.

**Comment by yurishkuro** [comment]
> Credits - we could change to full name, but so far I have seen using only handles as those do not change.

We can keep both. We used to do names only in the past:
```
- Use elasticsearch bulk API (#656) <Pavol Loffay>
- Support archive storage in the query-service (#604) <Yuri Shkuro>
- Introduce storage factory framework and composable CLI (#625) <Yuri Shkuro>
```

**Pull request merged**